grammars/go.cson

```diff
@@ -151,122 +229,255 @@
+      }
+    ]
   'delimiters':
     'patterns': [
       {
         'match': ','
-        'name': 'meta.delimiter.comma.go'
+        'name': 'punctuation.other.comma.go'
       }
       {
         'match': '\\.(?!\\.\\.)'
-        'name': 'meta.delimiter.period.go'
+        'name': 'punctuation.other.period.go'
       }
       {
         'match': ':(?!=)'
-        'name': 'meta.delimiter.colon.go'
+        'name': 'punctuation.other.colon.go'
+      }
+    ]
+  'keywords':
+    'patterns': [
+      {
+        'comment': 'Flow control keywords'
+        'match': '\\b(break|case|continue|default|defer|else|fallthrough|for|go|goto|if|range|return|select|switch)\\b'
+        'name': 'keyword.control.go'
+      }
+      {
+        'match': '\\bchan\\b'
+        'name': 'keyword.channel.go'
+      }
+      {
+        'match': '\\bconst\\b'
+        'name': 'keyword.constant.go'
+      }
+      {
+        'match': '\\bfunc\\b'
+        'name': 'keyword.function.go'
+      }
+      {
+        'match': '\\binterface\\b'
+        'name': 'keyword.interface.go'
+      }
+      {
+        'match': '\\bimport\\b'
+        'name': 'keyword.import.go'
+      }
+      {
+        'match': '\\bmap\\b'
+        'name': 'keyword.map.go'
+      }
+      {
+        'match': '\\bpackage\\b'
+        'name': 'keyword.package.go'
+      }
+      {
+        'match': '\\bstruct\\b'
+        'name': 'keyword.struct.go'
+      }
+      {
+        'match': '\\btype\\b'
+        'name': 'keyword.type.go'
+      }
+      {
+        'match': '\\bvar\\b'
+        'name': 'keyword.variable.go'
       }
     ]
   'operators':
+    'comment': 'Note that the order here is very important!'
     'patterns': [
       {
-        "name": "keyword.operator.go",
-        "match": "(\\+|&|\\+=|&=|&&|==|!=|\\-|\\||\\-=|\\|=|\\|\\||<|<=|\\*|\\^|\\*=|\\^=|<\\-|>|>=|/|<<|/=|<<=|\\+\\+|=|:=|%|>>|%=|>>=|\\-\\-|!|\\.\\.\\.|&\\^|&\\^=)"
+        'match': '(\\*|&)(?=\\w)'
+        'name': 'keyword.operator.address.go'
       }
       {
-        'match': '\\{|\\}'
-        'name': 'meta.brace.curly.go'
+        'match': '<\\-'
+        'name': 'keyword.operator.channel.go'
       }
       {
-        'match': '\\(|\\)'
-        'name': 'meta.brace.round.go'
+        'match': '\\-\\-'
+        'name': 'keyword.operator.decrement.go'
       }
       {
-        'match': '\\[|\\]'
-        'name': 'meta.brace.square.go'
+        'match': '\\+\\+'
+        'name': 'keyword.operator.increment.go'
+      }
+      {
+        'match': '(==|!=|<=|>=|<[^<]|>[^>])'
+        'name': 'keyword.operator.comparison.go'
+      }
+      {
+        'match': '(&&|\\|\\||!)'
+        'name': 'keyword.operator.logical.go'
+      }
+      {
+        'match': '(=|\\+=|\\-=|\\|=|\\^=|\\*=|/=|:=|%=|<<=|>>=|&\\^=|&=)'
+        'name': 'keyword.operator.assignment.go'
+      }
+      {
+        'match': '(\\+|\\-|\\*|/|%|\\||\\^|&\\^|&|<<|>>)'
+        'name': 'keyword.operator.arithmetic.go'
```

Comment:
I'm not sure about this one... Check this info for more details: https://golang.org/ref/spec#Arithmetic_operators

So I would radar keep it like it currently is, or call them (all bitwise related operators) `keyword.operator.arithmetic.bitwise.go`